### PR TITLE
Feature/hii 36

### PR DESCRIPTION
**Pull request by n3op2**
## What

adding a filter functionality along with the dropdown component.

#### changes
- new string prototype `.format()`  for key formatting
- filter
- search and filter functionality
- dynamic options pulled from geojson

#### screenshots
Screenshot 2023-01-12 at 13 33 03
Screenshot 2023-01-12 at 13 32 53

### CODE REVIEW

File: src/pages/Home.js [hunk 1]
```diff
@@ -35,10 +36,18 @@ const SearchWrapper = styled.div`
...
+  height: 100%;
+  align-content: start;
+  gap: 5px;
+  padding: 25px 15px;
+`
+
 const FullScreenGrid = styled(Grid)`
   height: 100vh;
   width: 100vw;
-  overflow: hidden; //TODO fix map overflow
```
Comment (jonmattgray) [inline src/pages/Home.js]: was `overflow: hidden` causing issues?

Comment (n3op2) [reply src/pages/Home.js]: sorry it was my bad removed thinking that this is invalid syntax while was looking for what was wrong with the formatting, it turned out to be something else, shared in the chat.

Comment (n3op2) [reply src/pages/Home.js]: updated

Comment (jonmattgray) [reply src/pages/Home.js]: it's not updated? no new commit

File: src/pages/Home.js [hunk 2]
```diff
@@ -69,6 +78,10 @@ const pointColourExpression = [
   '#27847A',
 ]
 
+String.prototype.format = function () {
+  return this.toLowerCase().replace(/\s/g, '_')
+}
```
Comment (mattdean-digicatapult) [inline src/pages/Home.js]: why are we changing string prototype?

**Review by mattdean-digicatapult — CHANGES_REQUESTED** [review]

File: src/pages/Home.js (hunk 2 above)
Comment (n3op2) [reply src/pages/Home.js]: adding a format() method, not really changing, I would say upgrading. To keep code cleaner. so I do not need to repeat this multiple times. Also creating a function i think it's an overkill, so added a new method to type String in this file. format()

Comment (n3op2) [reply src/pages/Home.js]: changed to a helper method.

**Review by mattdean-digicatapult — DISMISSED** [review]

**Pull request merged**